test/services/duplicate_rule_reference_resolver_test.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'test_helper'
+require './db/migrate/20200417174948_add_unique_index_to_rule_references'
+require 'sidekiq/testing'
+
+class DuplicateRuleReferenceResolverTest < ActiveSupport::TestCase
```

Comment:
New tests for the new resolvers were added. All the tests follow the same format.